### DISCUSSION TITLE
Database not loading

### DISCUSSION THREAD

**Discussion by RS2009** [simolus3/drift#2627] [Q&A] [answered]
```dart
LazyDatabase _openConnection() {
  return LazyDatabase(() async {
    final dbFolder = await getApplicationDocumentsDirectory();
    final file = File(path.join(dbFolder.path, 'Appdb.sqlite'));

    return NativeDatabase(file);
  });
}
```

I am using this code , but where exactly is the database getting stored?
I loaded some data using set function like setUsers etc,
When I converted my app to apk/run with a physical device no data is being displayed.
It is as if no data is present in file or the file itself is missing
and I am not getting the reason for that
I also tried insert on first app start using migration but it wasn't working.

**Comment by simolus3** [comment]
>  When I converted my app to apk/run with a physical device no data is being displayed.

Interesting. But it did work on a simulator, or did that not work either?

> I am using this code , but where exactly is the database getting stored?

On the device, using the path you're passing to `NativeDatabase`. You can use `print(file.absolute.path)` to print the path on the device/emulator.

**Comment by RS2009** [comment]
It is working with emulator

**Reply by simolus3** [reply]
Have you been able to find the database? I need more information to help debug this. Which Android version is this? Does the [example app](https://github.com/simolus3/drift/tree/develop/examples/app) work on the same device or does it have the same problem?

**Comment by RS2009** [comment, marked as answer]
Thanks it is working now
